data_sources/jolie/db_service.ol

```diff
@@ -95,22 +110,633 @@ main {
+                  .difficulty = result.row[ i ].difficulty;
+                  .place_of_origin = result.row[ i ].place_of_origin;
+                  .category = result.row[ i ].category;
+                  .cooking_technique = result.row[ i ].cooking_technique
+              }
+          }
+    }
+  }]
+
+
+  [ buildCommaSeparatedString( request )( response ) {
+        scope( sql ) {
+              install( SQLException => println@Console( sql.SQLException.stackTrace )();
+                                       throw( DatabaseError )
+              );
+
+              /* response = "," + request.str[0] + ","  */
+
+              println@Console(" Entered comma separated")();
+              response = ",";
+              for (idx = 0, idx < #request.str, idx++) {
+                response = response + request.str[idx] + ","
+              };
+              println@Console(" Generated comma separated: #" + response + "#")()
+
+        }
+  }]
+
+    [ buildList( request )( response ) {
+          scope( sql ) {
+                install( SQLException => println@Console( sql.SQLException.stackTrace )();
+                                         throw( DatabaseError )
+                );
+
+                response = "( ";
+                for (idx = 0, idx < #request.vector, idx++) {
+                  if (idx > 0) {
+                    response = response + ","
+                  };
+                  response = response + request.sep + request.vector[idx] + request.sep
+                };
+                response = response + " ) "
+          }
+    }]
+
+    [ buildIntList( request )( response ) {
+          scope( sql ) {
+                install( SQLException => println@Console( sql.SQLException.stackTrace )();
+                                         throw( DatabaseError )
+                );
+
+                response = "( ";
+                for (idx = 0, idx < #request.vector, idx++) {
+                  if (idx > 0) {
+                    response = response + ","
+                  };
+                  response = response + request.sep + request.vector[idx] + request.sep
+                };
+                response = response + " ) "
+          }
+    }]
+
+    [ buildSetVsSet( request )( response ) {
+          scope( sql ) {
+                install( SQLException => println@Console( sql.SQLException.stackTrace )();
+                                         throw( DatabaseError )
+                );
+
+                response = "( ";
+                for (idx = 0, idx < #request.vector, idx++ ) {
+                  if (idx > 0) {
+                      response = response + " OR "
+                    };
+                  matcher = request.sep + request.vector[idx] + request.sep;
+                  response = response + "( " + request.field + " LIKE '%" + matcher + "%' )"
+                };
+                response = response + " )"
+
+          }
+    }]
```

Comment:
queste operation di utilità sulle stringhe si potrebbero anche fare esternamente in javascript. Cmq ci guardo io alla prima occasione